src/Button/ButtonTitle.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native'; // TODO replace with our own text
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import StyleSheet from '../PlatformStyleSheet';
+
+import type { StylePropType } from '../PlatformStyleSheet/StyleTypes';
+import type { ButtonType } from './ButtonTypes';
+
+type Props = {|
+  +text: React.Node,
+  +type: ButtonType,
+  +style?: StylePropType,
+  +variant?: 'default' | 'sublabel',
+|};
+
+export default function ButtonTitle({
+  text,
+  style,
+  type = 'primary',
+  variant = 'default',
+}: Props) {
+  const variantStyle =
+    variant === 'sublabel' ? styles.sublabel : styles.default;
+  return (
+    <Text style={[styles.common, variantStyle, theme[type].text, style]}>
+      {text}
+    </Text>
+  );
+}
+
+const styles = StyleSheet.create({
+  common: {
+    android: {
+      textTransform: 'uppercase',
+    },
+  },
+  default: {
+    fontWeight: 'bold',
+    fontSize: 16,
+    lineHeight: 21,
+  },
+  sublabel: {
+    fontSize: 14,
+    lineHeight: 18,
+    fontWeight: 'normal',
+  },
+});
+
+const theme = {
+  primary: StyleSheet.create({
+    text: {
+      color: defaultTokens.colorTextButtonPrimary,
+    },
+  }),
```

Comment:
As there is lot of code, I feel we can save some in several places. I have the feeling you can have just one style for this:

```
const theme = (type) => {
  text: StyleSheet.create({
    color: defaultTokens[`colorTextButton${type.charAt(0).toUpperCase()}${type.slice(1)}`],
  }),
}
```

Reply:
I would like to use this approach, but sometimes the colors we need to apply for RN do not follow exactly the defaultTokens nomenclature. That's why I had to add `web` for these few exceptions. But I will try to shorten it.